src/tests/base_test.py

```diff
@@ -141,6 +173,28 @@ def test_something(cluster):
         """
         yield utils.run_marked_fixture(prepared_controller_configuration, "override_controller_configuration", request)
 
+    @pytest.fixture
+    def day2_controller_configuration(
+        self, request: pytest.FixtureRequest, prepared_day2_controller_configuration: BaseNodesConfig
+    ) -> BaseNodesConfig:
+        """
+        Allows the test to modify the controller configuration by registering a custom fixture.
+        To register the custom fixture you have to mark the test with "override_controller_configuration" marker.
+
+        For example:
+
+        @pytest.fixture
+        def FIXTURE_NAME(self, prepared_controller_configuration):
+            yield prepared_controller_configuration
+
+        @pytest.mark.override_controller_configuration(FIXTURE_NAME.__name__)
+        def test_something(cluster):
+            pass
```

Comment:
not an easy API
but at least there's an example usage now, so I'm grateful for that :)

Reply:
it was already there in `controller_configuration()`  :)